test/models/ride_test.dart

```diff
@@ -0,0 +1,961 @@
+      expect(json.keys.length, 18);
+    });
+  });
+  group('Ride.userHasRideAtTimeRange', () {
+    test('returns true when there is a approved not finished Ride at the time range', () async {
+      when.call(rideProcessor.processUrl(any)).thenReturn(jsonEncode([
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  status: RideStatus.approved,
+                  startTime: DateTime.now().add(const Duration(hours: 2)),
+                  endTime: DateTime.now().add(const Duration(hours: 4)),
+                )
+                .toJsonForApi()
+          ]));
+      expect(
+          await Ride.userHasRideAtTimeRange(
+              DateTimeRange(start: DateTime.now(), end: DateTime.now().add(const Duration(hours: 10))), 2),
+          true);
+    });
+    test('returns false if there is no approved and not finished ride at the time range', () async {
+      when.call(rideProcessor.processUrl(any)).thenReturn(jsonEncode([
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  status: RideStatus.pending,
+                  startTime: DateTime.now().add(const Duration(hours: 8)),
+                  endTime: DateTime.now().add(const Duration(hours: 10)),
+                )
+                .toJsonForApi(),
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 2)),
+                  endTime: DateTime.now().add(const Duration(hours: 4)),
+                )
+                .toJsonForApi(),
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  startTime: DateTime.now().subtract(const Duration(hours: 4)),
+                  endTime: DateTime.now().subtract(const Duration(hours: 2)),
+                )
+                .toJsonForApi(),
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  status: RideStatus.approved,
+                  startTime: DateTime.now().add(const Duration(hours: 16)),
+                  endTime: DateTime.now().add(const Duration(hours: 20)),
+                )
+                .toJsonForApi(),
+            RideFactory()
+                .generateFake(
+                  riderId: 2,
+                  startTime: DateTime.now().add(const Duration(hours: 12)),
+                  endTime: DateTime.now().add(const Duration(hours: 15)),
+                  status: RideStatus.cancelledByDriver,
+                )
+                .toJsonForApi(),
+          ]));
+      expect(
+          await Ride.userHasRideAtTimeRange(
+              DateTimeRange(
+                start: DateTime.now().add(const Duration(hours: 4)),
+                end: DateTime.now().add(const Duration(hours: 15)),
+              ),
+              2),
+          false);
+    });
+  });
+  group('Ride.shouldShowInListView', () {
```

Comment:
do we really need to test these methods for ride and drive?

Reply:
I also think that they should be moved to some trip_test.
 
But as a note: There you can only use a model of a specific type... So if I tested only for ride in that file, then I would not catch it if the behaviour were to be changed for drives in the future.